### PR TITLE
Alert feature

### PR DESCRIPTION
**Pull request by milosyuki**

### CODE REVIEW

File: src/app/(static-cms)/cms/config.ts [hunk 1]
```diff
@@ -438,6 +438,54 @@ const blocks: ListField["types"] = [
...
+          },
+          {
+            name: "children",
+            label: "Description",
+            widget: "markdown",
+          }
+        ]
+      }
+    ]
+  }
```
Comment (thecotne) [inline src/app/(static-cms)/cms/config.ts:440]: this should be removed right?

File: src/app/(static-cms)/cms/config.ts [hunk 2]
```diff
@@ -593,6 +641,12 @@ export const config: Config = {
           widget: "boolean",
           default: true,
         },
+        {
+          name: "alert",
+          label: "Show alert",
+          widget: "boolean",
+          default: false,
+        },
```
Comment (thecotne) [inline src/app/(static-cms)/cms/config.ts]: this should be removed right?

File: src/app/(static-cms)/cms/config.ts [hunk 3]
```diff
@@ -647,6 +701,54 @@ export const config: Config = {
...
+                },
+                {
+                  name: "children",
+                  label: "Description",
+                  widget: "markdown",
+                }
+              ]
+            }
+          ]
+        },
```
Comment (thecotne) [inline src/app/(static-cms)/cms/config.ts]: this should be removed right?

File: src/app/[locale]/(components)/PageContainer.tsx [hunk 4]
```diff
@@ -1,14 +1,36 @@
...
+
   return (
     <Flex direction="column" flex="1">
+      {result ? <Alert title={result.title} hasCloseButton={result.hasCloseButton} uuid={result.id} variant={result.variant}>{result.children}</Alert> : alerts?.map((alert, i) => {
+        if (!alert.page_url && typeof window !== 'undefined') {
+          if (!window.localStorage.getItem(`uuid-${alert.id}`)) {
+            return <Alert key={`${i}-alert`} title={alert.title} hasCloseButton={alert.hasCloseButton} uuid={alert.id} variant={alert.variant}>{alert.children}</Alert>
+          }
+        }
+      })}
```
Comment (thecotne) [inline src/app/[locale]/(components)/PageContainer.tsx]: ~~can u run `yarn fmt` to format code~~

File: workspaces/netlify-cms/src/config.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import {
   CmsFieldList,
   CmsFieldMeta,
 } from "netlify-cms-core";
+import { icons } from 'react-icons/lib/cjs';
```
Comment (thecotne) [inline workspaces/netlify-cms/src/config.ts]: this is not used right?

File: workspaces/netlify-cms/src/config.ts [hunk 6]
```diff
@@ -425,6 +426,42 @@ const blocks: CmsFieldList["types"] = [
...
+          },
+          {
+            name: "children",
+            label: "Description",
+            widget: "markdown",
+          }
+        ]
+      }
+    ]
+  }
```
Comment (thecotne) [inline workspaces/netlify-cms/src/config.ts:427]: this is unused

File: workspaces/netlify-cms/src/config.ts [hunk 7]
```diff
@@ -621,6 +658,61 @@ export const config: CmsConfig = {
...
+                  name: "page_url",
+                  label: "Page url",
+                  hint: "If page url is not specified (e.g. 'learn/glossary'), it will be used globally",
+                  required: false,
+                  widget: "string"
+                },
+              ]
+            }
+          ]
+        },
```
Comment (thecotne) [inline workspaces/netlify-cms/src/config.ts]: this is also unused

File: workspaces/content-onedge/src/data.ts [hunk 8]
```diff
@@ -82,6 +82,7 @@ export interface Page extends Meta {
   readonly template: "landing" | "content";
   readonly breadcrumbs: boolean;
   readonly pageLastUpdated: boolean;
+  readonly alert: boolean;
```
Comment (thecotne) [inline workspaces/content-onedge/src/data.ts]: this is unused

File: workspaces/content-onedge/src/data.ts [hunk 9]
```diff
@@ -118,6 +119,7 @@ export async function fileToPage(
     blocks: data.blocks ?? [],
     id: data.id,
     parent_page: data.parent_page,
+    alert: data.alert,
```
Comment (thecotne) [inline workspaces/content-onedge/src/data.ts]: this is unused

File: workspaces/content-onedge/src/data.ts [hunk 10]
```diff
@@ -375,6 +377,7 @@ export function handleLink(
         template: data.template,
         breadcrumbs: data.breadcrumbs,
         pageLastUpdated: data.pageLastUpdated,
+        alert: data.alert,
```
Comment (thecotne) [inline workspaces/content-onedge/src/data.ts]: this is unused

File: src/data/pages.ts [hunk 11]
```diff
@@ -185,6 +195,7 @@ export interface Page extends Meta {
   readonly breadcrumbs_data?: readonly Omit<Page, "blocks">[];
   readonly pageLastUpdated: boolean;
   readonly page_last_updated?: string;
+  readonly alert?: Alert;
```
Comment (thecotne) [inline src/data/pages.ts]: this is unused

File: src/data/pages.ts [hunk 12]
```diff
@@ -175,6 +176,15 @@ export interface GroupBlock {
...
 export type TopLevelBlock = Block | FlexLayoutBlock | GroupBlock | Container;
 
+export interface Alert {
+  title: string;
+  hasCloseButton: boolean;
+  id: string;
+  variant: 'important' | 'info' | 'warning';
+  children: ReactNode;
+  page_url: string;
+}
```
Comment (thecotne) [inline src/data/pages.ts]: this is unused

File: src/data/settings/alert.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { defaultLocale } from "../i18n/config";
+import { getFirst, getJSON } from "../utils";
+
+export interface Alert {
+  readonly title: string;
+  readonly children: React.ReactNode;
+  readonly toId?: string;
```
Comment (thecotne) [inline src/data/settings/alert.ts]: what's this field?

**Review by thecotne — CHANGES_REQUESTED** [review]

**Comment by thecotne** [comment]
@milosyuki looks good 👍 we just need to remove some left over code)

File: src/app/[locale]/(components)/PageContainer.tsx (hunk 4 above)
Comment (thecotne) [reply src/app/[locale]/(components)/PageContainer.tsx]: seems like we were forgetting to format code and formatting all files will introduce too many changes in this pr

just format this file

File: src/app/(static-cms)/cms/config.ts (hunk 2 above)
Comment (milosyuki) [reply src/app/(static-cms)/cms/config.ts]: Removed, thanks.

File: src/app/(static-cms)/cms/config.ts (hunk 3 above)
Comment (milosyuki) [reply src/app/(static-cms)/cms/config.ts]: Removed, thanks.

File: workspaces/netlify-cms/src/config.ts (hunk 5 above)
Comment (milosyuki) [reply workspaces/netlify-cms/src/config.ts]: Removed, thanks.

File: workspaces/netlify-cms/src/config.ts (hunk 7 above)
Comment (milosyuki) [reply workspaces/netlify-cms/src/config.ts]: Removed, thanks.

File: workspaces/content-onedge/src/data.ts (hunk 8 above)
Comment (milosyuki) [reply workspaces/content-onedge/src/data.ts]: Removed, thanks.

File: workspaces/content-onedge/src/data.ts (hunk 9 above)
Comment (milosyuki) [reply workspaces/content-onedge/src/data.ts]: Removed, thanks.

File: workspaces/content-onedge/src/data.ts (hunk 10 above)
Comment (milosyuki) [reply workspaces/content-onedge/src/data.ts]: Removed, thanks.

File: src/data/pages.ts (hunk 11 above)
Comment (milosyuki) [reply src/data/pages.ts]: Removed, thanks.

File: src/data/pages.ts (hunk 12 above)
Comment (milosyuki) [reply src/data/pages.ts]: Removed, thanks.

File: src/data/settings/alert.ts (hunk 13 above)
Comment (milosyuki) [reply src/data/settings/alert.ts]: Here by mistake. Removed, thanks.

**Review by thecotne — APPROVED** [review]

**Pull request merged**